grails-app/services/org/pih/warehouse/product/ProductService.groovy

```diff
@@ -1284,6 +1310,14 @@ class ProductService {
                 }
             }
         }
+        if(isNewInstance){ // Saving product id in product attribute
+            Attribute attribute = findExternalProductIdAttribute()
```

Comment:
Let's add this as a static method to Attribute if we can.
```
Attribute.getExternalProductIdAttribute()
```

Reply:
if we add static then we cannot use grailsApplication service.